packages/info/mixins/doctor/mixin.core.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const {
+  async: { parallel },
+} = require('mixinable');
+
+const {
+  Mixin,
+  internal: { validate, invariant },
+} = require('@untool/core');
+
+const { createDoctor } = require('../../lib/doctor');
+
+class DoctorMixin extends Mixin {
+  constructor(...args) {
+    super(...args);
+    this.doctor = createDoctor(this.config);
+  }
+  diagnose({ diagnoseConfig, diagnoseDuplicatePackages }) {
+    diagnoseConfig();
+    diagnoseDuplicatePackages('@untool/*');
+  }
+  bootstrap() {
+    const { doctor } = this;
+    return this.diagnose(doctor).then((results) =>
+      doctor.submitDiagnosis(
+        ...[].concat(...results.filter((result) => result !== undefined))
```

Comment:
Why can't we just write `...results.filter((result) => result !== undefined)`?

Reply:
Because we must expect the individual results to arrays, so we would get an array of arrays if we did what you propose. We need to flatten that.